service/header/store.go

```diff
@@ -183,9 +184,13 @@ func (s *store) Append(ctx context.Context, headers ...*ExtendedHeader) error {
 
 	verified := make([]*ExtendedHeader, 0, lh)
 	for _, h := range headers {
-		if head.Height == h.Height {
+		if head.Height == h.Height && bytes.Equal(head.Hash(), h.Hash()) {
+			log.Warnw("duplicate header", "hash", head.Hash())
 			continue
 		}
+		// TODO(@Wondertan): Fork choice rule - if two headers are on the same height, but have different hashes
+		//  choose one who has more consensus(power) over it.
+		//  It is not critical to implement this rn, because we can reliably apply the last header who has +2/3.
```

Comment:
Shouldn't we halt if there's two headers with a Tendermint Commit within the unbonding window? And if it's outside the unbonding window, you should just use the one that's an ancestor of the trusted hash?

Reply:
Ok, so we need centralized halting management in the node that can be triggered by multiple places: header verification and fraud proofs.

Reply:
Btw, I created [this issue](https://github.com/celestiaorg/celestia-node/issues/235) during devnet as per @adlerjohn's suggestion. Should I close it now?

Reply:
[An issue ](https://github.com/celestiaorg/celestia-node/issues/365)for halting.

Reply:
Looks like this second issue supersedes the first, so sure.